### PR TITLE
Add domain blacklist

### PR DESCRIPTION
**Pull request by fantasticmrdavid**
Trello: https://trello.com/c/9qZloYGH/434-fe-swap-out-parsely-plugin-for-js-and-exclude-paid-sources

### CODE REVIEW

File: parsely-referrer-blacklist.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "googlesyndication",
+  "74.55.82.101",
+  "webmonetizer",
+  "traffic2money",
+  "boost-my-site",
+  "buyfastwebtraffic",
+  "qobk"
+);
+
+$blacklist = array_map('strtolower', $blacklist);
```
Comment (santouras) [inline parsely-referrer-blacklist.php]: this `strtolower` call is redundant given they are all stored in lowercase.....?

File: wp-parsely.php [hunk 2]
```diff
@@ -392,12 +392,22 @@ public function insert_parsely_javascript() {
         $parselyOptions = $this->get_options();
         // If we don't have an API key, there's no need to proceed.
         if ( empty($parselyOptions['apikey']) ) {
-            return '';
+            //return '';
```
Comment (santouras) [inline wp-parsely.php]: left over from testing or is it intentional to continue w/o an API key?

File: wp-parsely.php [hunk 3]
```diff
@@ -392,12 +392,22 @@ public function insert_parsely_javascript() {
...
 
         global $post;
         $display = TRUE;
 
+        $referrer = isset($_SERVER['HTTP_REFERER']) ? $_SERVER['HTTP_REFERER'] : "";
+        include_once('parsely-referrer-blacklist.php');
+        $blacklisted = false;
+
+        foreach ( $blacklist as $str ) {
+          $blacklisted = true && strpos($referrer, $str) !== false;
```
Comment (santouras) [inline wp-parsely.php]: the `true &&` here does nothing

**Comment by fantasticmrdavid** [comment]
Updated

**Comment by mrlagmer** [comment]
Looks good.

**Comment by santouras** [comment]
👍

**Pull request merged**